include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BATTERY
```

Comment:
Uh... why? Don't think any other subsystem does this, can't think of a good reason to silently blank the file.

Reply:
Good catch, pretty sure that was for early testing.